### PR TITLE
Improve Tide's queries details on Tide page and PR Dashboard

### LINKED ISSUE #9335: tide status page does not indicate milestone requirements
We use two k/k queries that have different milestone requirements, but they look as though they're the same query in the UI

screen shot 2018-09-11 at 10 04 05 am

/area prow
/kind bug
/priority important-soon

### LINKED ISSUE #9361: tide status page does not render org queries properly
the logic currently only handles org/repo queries, you get an empty bullet point for org level queries. we should fix this

/area prow/deck
/kind bug
cc @amwat 
related to: https://github.com/kubernetes/test-infra/issues/9335
/priority important-soon

### PR DESCRIPTION
**Pull request by qhuynh96**
* Fixes PR & tide query matching. The matching function should take `includedBranches` and `excludedBranches` into account.
* Add information about `milestone`, `includedBranches`, `excludedBranches` to PR Dashboard Query detail & Tide page
* Minor UI treatment
* Fixes #9335
* Fixes #9361
* Screenshots:
screen shot 2018-09-13 at 7 07 16 pm

screen shot 2018-09-13 at 7 06 57 pm

screen shot 2018-09-14 at 10 18 12 am

/cc @BenTheElder 
/cc @stevekuznetsov 
/cc @amwat

### CODE REVIEW

**Comment by BenTheElder** [comment]
> Changes seem good. I wonder if we can do better at an architectural level to stop needing to null-check everything in the JS

Use another language? ;-) There are a number of transpile ones we could have fun with, some of them are pretty nice.

**Comment by BenTheElder** [comment]
hmm merge conflict (needs rebase), mind posting what this looks like for the org-wide queries? this looks really nice :-)

**Comment by BenTheElder** [comment]
(it's also really handy to be able to go back through github PRs / issues and show the incremental improvements without having to try to get them up and running for each version...)

File: prow/cmd/deck/static/pr.html [hunk 1]
```diff
@@ -76,9 +76,21 @@ <h4 class="mdl-dialog__title"></h4>
...
+        <div id="query-detail-milestone" class="hidden">
+            <p class="detail-title">Milestone</p>
+            <p class="detail-data"></p>
+        </div>
+        <div id="query-detail-exclude" class="hidden">
+            <p class="detail-title">Exclude branches</p>
+            <p class="detail-data"></p>
+        </div>
+        <div id="query-detail-include" class="hidden">
+            <p class="detail-title">Include branches</p>
```
Comment (BenTheElder) [inline prow/cmd/deck/static/pr.html]: possibly "included" and "excluded" for these? to be consistent with required and forbidden

Comment (qhuynh96) [reply prow/cmd/deck/static/pr.html]: done +1

File: prow/cmd/deck/static/tidescript.js [hunk 2]
```diff
@@ -129,63 +153,54 @@ function redrawQueries() {
...
+        // list milestone if existed
+        fillDetail(tideQuery.milestone, "milestone", "with ", li, function(data) {
+            return document.createTextNode(data);
+        });
+        // list all excluded branches
+        fillDetail(tideQuery.excludedBranches, "branches", "exclude ", li, function(data) {
+            return document.createTextNode(data);
+        });
+        // list all included branches
+        fillDetail(tideQuery.includedBranches, "branches", "effect on ", li, function(data) {
```
Comment (stevekuznetsov) [inline prow/cmd/deck/static/tidescript.js]: Made against branches?
Merging into branches?
Targetting branches?

Comment (qhuynh96) [reply prow/cmd/deck/static/tidescript.js]: I chose targeting 👍

**Pull request merged**